services/minespace-web/src/components/dashboard/mine/reports/ReportPage.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, useParams } from "react-router-dom";
+import { getFormSubmitErrors, getFormValues, submit } from "redux-form";
+
+import { Button, Col, Row, Typography } from "antd";
+import ArrowLeftOutlined from "@ant-design/icons/ArrowLeftOutlined";
+
+import { FORM } from "@mds/common/constants/forms";
+import ReportDetailsForm from "@mds/common/components/reports/ReportDetailsForm";
+import { fetchMineReport } from "@mds/common/redux/actionCreators/reportActionCreator";
+import { getMineReportById } from "@mds/common/redux/reducers/reportReducer";
+import Loading from "@/components/common/Loading";
+import { getMineById } from "@mds/common/redux/selectors/mineSelectors";
+import { fetchMineRecordById } from "@mds/common/redux/actionCreators/mineActionCreator";
+import { IMineReport, MINE_REPORT_STATUS_HASH, MINE_REPORT_SUBMISSION_CODES } from "@mds/common";
+import Callout from "@mds/common/components/common/Callout";
+import { reportStatusSeverity } from "./ReportsTable";
+
+const ReportPage = () => {
+  const dispatch = useDispatch();
+  const { mineGuid, reportGuid } = useParams<{ mineGuid: string; reportGuid: string }>();
+  const mineReport = useSelector((state) => getMineReportById(state, reportGuid));
+  const mine = useSelector((state) => getMineById(state, mineGuid));
+  const [loaded, setIsLoaded] = useState(mineReport && mine);
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  // get form data so we can submit it outside of the form
+  const formErrors = useSelector(getFormSubmitErrors(FORM.VIEW_EDIT_REPORT));
+  const formValues = useSelector(getFormValues(FORM.VIEW_EDIT_REPORT));
+
+  useEffect(() => {
+    if (!mine || mine.mine_guid !== mineGuid) {
+      dispatch(fetchMineRecordById(mineGuid));
+    }
```

Comment:
Can the logic here can be moved to the other useEffect on line 38 since the useEffect gets triggered when either reportGuid or mineGuid changes ?